app/models/preview_card.rb

```diff
@@ -50,7 +50,9 @@ class PreviewCard < ApplicationRecord
   enum type: { link: 0, photo: 1, video: 2, rich: 3 }
   enum link_type: { unknown: 0, article: 1 }
 
-  has_and_belongs_to_many :statuses
+  has_many :preview_cards_statuses, dependent: :destroy
```

Comment:
Wouldn't `:delete_all` be more fitting? I'm afraid `:destroy` could have quite some overhead here.